examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/circuitInputsParser/circuitInputParser.service.ts

```diff
@@ -5,6 +5,59 @@ import { LoggingService } from '../../../../../shared/logging/logging.service';
 export class CircuitInputsParserService {
   constructor(private readonly logger: LoggingService) {}
 
+  public validateCircuitInputTranslationSchema(
+    schema: string,
+  ): [boolean, string] {
+    try {
+      const parsedData: CircuitInputsMapping = JSON.parse(schema);
+
+      if (!parsedData.mapping || !Array.isArray(parsedData.mapping)) {
+        return [false, `Missing mapping array`];
+      }
+
+      for (const mapping of parsedData.mapping) {
+        if (typeof mapping.circuitInput !== 'string') {
+          return [false, `${mapping.circuitInput} not of type string`];
+        }
+
+        if (typeof mapping.description !== 'string') {
+          return [false, `${mapping.description} not of type string`];
+        }
+
+        if (typeof mapping.payloadJsonPath !== 'string') {
+          return [false, `${mapping.payloadJsonPath} not of type string`];
+        }
+
+        if (typeof mapping.dataType !== 'string') {
+          return [false, `${mapping.dataType} not of type string`];
+        }
+
+        if (mapping.dataType === 'array') {
+          if (!mapping.arrayType || typeof mapping.arrayType !== 'string') {
+            return [
+              false,
+              `arrayType not defined properly for ${mapping.circuitInput}`,
+            ];
+          }
+        }
+
+        if (
+          mapping.defaultValue &&
+          typeof mapping.defaultValue !== mapping.dataType
+        ) {
+          return [
+            false,
+            `defaultValue not of type ${mapping.dataType} for ${mapping.circuitInput}`,
+          ];
+        }
+      }
+
+      return [true, ''];
```

Comment:
i dont think we should return both boolean and error, looks kinda weird in javascript, lets just return error which implies that validation is true or false? but up to you

Reply:
i ike it

Reply:
@ognjenkurtic are you going to leave it like that? L57 still returns a boolean and error message. Same with L55 ... I would return a No-Error Message

Reply:
Fixed